### DISCUSSION TITLE
questions about run_8point

### DISCUSSION THREAD

**Discussion by zh7117** [kornia/kornia#2903] [Questions] [answered]
Hello everyone. 
I found the implementation of algorithm [run_8point](https://github.com/kornia/kornia/blob/e34ed3c51cdd9a96ad9e6e03b50e4ff8a8aa3319/kornia/geometry/epipolar/fundamental.py#L225) is slightly different from [wiki-8point](https://en.wikipedia.org/wiki/Eight-point_algorithm#:~:text=The%20eight%2Dpoint%20algorithm%20is,case%20of%20the%20essential%20matrix.)
In kornia: 
```
# apply the weights to the linear system
if weights is None:
    X = X.transpose(-2, -1) @ X
else:
    w_diag = torch.diag_embed(weights)
    X = X.transpose(-2, -1) @ w_diag @ X

_, _, V = _torch_svd_cast(X)
F_mat = V[..., -1].view(-1, 3, 3)

# reconstruct and force the matrix to have rank2
U, S, V = _torch_svd_cast(F_mat)
rank_mask = torch.tensor([1.0, 1.0, 0.0], device=F_mat.device, dtype=F_mat.dtype)

F_projected = U @ (torch.diag_embed(S * rank_mask) @ V.transpose(-2, -1))
```
But according wiki, there is no need to multiply X by its own transpose.

**Comment by edgarriba** [comment, marked as answer]
i believe this is a trick to normalize the matrix before the singular values decomposition so that is squared /cc @ducha-aiki

**Reply by zh7117** [reply]
Thanks for your response. 
I found wherther or not multiply the X^T does not affect the value of V[...,-1]. 
So this would be a trick to reduce computations or something else.

**Reply by edgarriba** [reply]
yes, it's a performance trick
